.circleci/config.yml

```diff
@@ -32,7 +34,7 @@ jobs:
       - run: sudo ./scripts/install-protoc.sh
       - run: ./scripts/for-each-module.sh "go generate ./..."
       - run: make vet check
-      - run: ./scripts/for-each-module.sh "go build ./..."
+      - run: sudo sudo chmod -R guo+w /usr/local/go/pkg
```

Comment:
This seems related to my error:
```
go build vendor/golang.org/x/crypto/internal/subtle: mkdir /usr/local/go/pkg/linux_amd64: permission denied
```

Is there a way to avoid needing to munge system level permissions?  Why are we having to do so?

Reply:
Yes, put GOROOT not into system location fixes this. And I suppose this is only relevant for macos

Reply:
I still don't understand why we have this issue to begin with... why are we munging with GOROOT?

Reply:
OK... figured it out.  If you don't use -i you don't have this problem.  In fact its [recommended not to use -i anymore at all](https://github.com/golang/go/issues/24674#issuecomment-379143645)